pkg/util/winutil/iisconfig/apmtags_test.go

```diff
@@ -0,0 +1,117 @@
+)
+
+func TestAppConfig(t *testing.T) {
+	path, err := os.Getwd()
+	require.Nil(t, err)
+
+	apppath := filepath.Join(path, "testdata", "app_1.config.xml")
+	//apppath := filepath.Join(path, "testdata", "iisconfig.xml")
+	iisCfg, err := ReadDotNetConfig(apppath)
+	assert.Nil(t, err)
+	assert.NotNil(t, iisCfg)
+
+	assert.Equal(t, iisCfg.DDService, "service1")
+	assert.Equal(t, iisCfg.DDEnv, "false")
+	assert.Equal(t, iisCfg.DDVersion, "1.0-prerelease")
+
+}
+
+func TestPathSplitting(t *testing.T) {
+
+	t.Run("Test Root path", func(t *testing.T) {
+		sp := splitPaths("/")
+		assert.Equal(t, 0, len(sp))
+	})
+
+	t.Run("Test path depth 3", func(t *testing.T) {
+		sp := splitPaths("/path/to/app")
+		assert.Equal(t, 3, len(sp))
+		assert.Equal(t, "path", sp[0])
+		assert.Equal(t, "to", sp[1])
+		assert.Equal(t, "app", sp[2])
+	})
+
+	t.Run("Test path depth 3 with trailing slash", func(t *testing.T) {
+		sp := splitPaths("/path/to/app/")
+		assert.Equal(t, 3, len(sp))
+		assert.Equal(t, "path", sp[0])
+		assert.Equal(t, "to", sp[1])
+		assert.Equal(t, "app", sp[2])
+	})
+
+}
+
+func TestAPMTags(t *testing.T) {
+	path, err := os.Getwd()
+	require.Nil(t, err)
+
+	iisCfgPath = filepath.Join(path, "testdata", "apptest.xml")
+	testroot := filepath.Join(path, "testdata")
+	os.Setenv("TESTROOTDIR", testroot)
+	defer os.Unsetenv("TESTROOTDIR")
+
+	//apppath := filepath.Join(path, "testdata", "iisconfig.xml")
+	iisCfg, err := NewDynamicIISConfig()
+	assert.Nil(t, err)
+	assert.NotNil(t, iisCfg)
+
+	err = iisCfg.Start()
+	assert.Nil(t, err)
+
+	t.Run("Test simple root path", func(t *testing.T) {
+		tags, _ := iisCfg.GetAPMTags(2, "/")
+		assert.Equal(t, "app1", tags.DDService)
+	})
+	t.Run("Test deeper path from top app", func(t *testing.T) {
+		tags, _ := iisCfg.GetAPMTags(2, "/path/to/app")
+		assert.Equal(t, "app1", tags.DDService)
+	})
+
+	t.Run("test top level app2", func(t *testing.T) {
+		tags, _ := iisCfg.GetAPMTags(2, "/app2")
+		assert.Equal(t, "app2", tags.DDService)
+	})
+	t.Run("test deeper app2", func(t *testing.T) {
+		tags, _ := iisCfg.GetAPMTags(2, "/app2/some/path")
+		assert.Equal(t, "app2", tags.DDService)
+	})
+
+	t.Run("test app3 nested in app2", func(t *testing.T) {
+		tags, _ := iisCfg.GetAPMTags(2, "/app2/app3")
+		assert.Equal(t, "app3", tags.DDService)
+	})
+	t.Run("test app3 nested in app2 with path", func(t *testing.T) {
+		tags, _ := iisCfg.GetAPMTags(2, "/app2/app3/some/path")
+		assert.Equal(t, "app3", tags.DDService)
+	})
+
+	t.Run("test secondary site", func(t *testing.T) {
+		tags, _ := iisCfg.GetAPMTags(3, "/")
+		assert.Equal(t, "app1", tags.DDService)
+	})
+	t.Run("test secondary site app 3", func(t *testing.T) {
+		// this should still be app1 because the root path on the
+		// second site is different
+		tags, _ := iisCfg.GetAPMTags(3, "/app2/app3")
+		assert.Equal(t, "app1", tags.DDService)
+	})
+	t.Run("test secondary site actual app3", func(t *testing.T) {
+		tags, _ := iisCfg.GetAPMTags(3, "/siteapp2/siteapp3")
```

Comment:
Does it matter if the path is to a file?

Reply:
shouldn't, but I added that case.